### PR TITLE
new site

### LINKED ISSUE #1013: Get rid of documentation versioning
We don't want to maintain a non-latest version of the [documentation](https://remark42.com/docs/), I propose to get rid of v1.6 and latest and make it work only as single version.

### PR DESCRIPTION
**Pull request by akellbl4**
I've redeveloped the site with Eleventy. It's an alternative for Hugo with a much friendly API and DX.
Finally, we've got working navigation in docs.

[Live demo](https://remark42.vercel.app/)

Fixes #1013

- [x] implement 404 page
- [x] implement theme switcher
- [x] implement dark theme
- [x] add code highlighting
- [x] mobile version for docs
- [x] fix styles for a dark theme when having all of the content

Maybe features down below better do in separate PR and release new site sooner.

- add links on next/prev pages at the end of doc pages
- add a table of content
- implement a page with a list of the releases
- add copy button on code blocks

<details>
<summary>Screenshots</summary>

CleanShot 2021-06-12 at 15 35 14@2x
CleanShot 2021-06-12 at 15 35 11@2x
CleanShot 2021-06-12 at 15 19 26@2x

CleanShot 2021-06-13 at 11 10 34@2x
CleanShot 2021-06-13 at 11 10 37@2x
CleanShot 2021-06-13 at 11 10 40@2x
</details>

### CODE REVIEW

**Comment by akellbl4** [comment]
@paskal I've created a structure of content. If you have any ideas we can change it.

File: site/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+node_modules
```
Comment (paskal) [inline site/.dockerignore:1]: /site should have its own Readme.md

File: site/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM node:14-alpine
+
+WORKDIR site
+
+COPY ./ /site
+COPY ../docs /site/src/docs
```
Comment (paskal) [inline site/Dockerfile:6]: I can't build it, unfortunately:

```
 > [4/5] COPY ../docs /site/src/docs:
------
failed to compute cache key: "/docs" not found: not found
```
Also, I don't know what would be the result given the fact that the run CMD sleeps.

**Review by paskal — CHANGES_REQUESTED** [review]

**Comment by akellbl4** [comment]
@paskal thank you for help!

**Comment by paskal** [comment]
Really nice site! Few things I found so far:

- Main page has extra empty `<li>` element
  image
- /docs/system-requirments/ should be /docs/system-requirements/, also edit button leads to `https://github.com/umputun/remark42/edit//src/docs/getting-started/system-requirments/index.md` with two slashes, one is redundant
- in my case, `docker-compose run build` generates the site properly, but `docker-compose up server` serves the site without style.css
- Right now the site generated with `docker-compose run build` is empty, current documentation is not included: what I see is plenty of empty pages, /docs/contributing/technical-details/ is the only one I found with text.

**Comment by akellbl4** [comment]
But> but docker-compose up server serves the site without style.css

Strange, works good for me.

> Right now, the site generated with docker-compose run build is empty, current documentation is not included: what I see is plenty of empty pages, /docs/contributing/technical-details/ is the only one I found with text.

I've created proper structure for docs but haven't moved them to the site yet.
If anybody can help with it would be great.

> code on the only page with text I have is wrapped in apostrofs

They are backticks and it's intentional. I use default styles for it from tailwind.

> The 404 page you designed is not shown on the yarn dev run. For consistency, I would propose fixing that as well.

As far as I know dev server doesn't support custom 404 and it can be reached by open `/404/`. In production configuration should refer to that page for displaying 404

**Comment by akellbl4** [comment]
> - Main page has extra empty `<li>` element
>  image

I've fixed this bug. It was bad support for the dark theme on the main page.

**Comment by akellbl4** [comment]
Live demo:
https://remark42.vercel.app/

**Comment by akellbl4** [comment]
> /docs/system-requirments/ should be /docs/system-requirements/, also edit button leads to https://github.com/umputun/remark42/edit//src/docs/getting-started/system-requirments/index.md with two slashes, one is redundant

**Comment by akellbl4** [comment]
@paskal try to run the server in docker. looks like I've fixed it.

**Comment by akellbl4** [comment]
@paskal @umputun 
I've done everything from the code perspective. Feel free to test and comment.
I think this PR is good to merge. I created a basic structure for docs to demonstrate how navigation works. We can do all of the work on migration docs in a separate PR.

**Review by paskal — APPROVED** [review]

**Pull request merged**